### ISSUE TITLE
FR: Folder to ignore Examples and Functional Extensions Needed

### ISSUE THREAD

**Issue by mineclover** [platers/obsidian-linter#581]
An example of Folder to ignore is needed.
Also, it would be nice to set a folder that does not lint when

**Comment by mineclover** [comment]
For example I don't want linting to happen where I use it as a template folder.

But when I run full file lint
Lint also runs inside the generated template folder.

Handling exceptions in tagged template files is also not a good idea.
Templates are created so that they can be applied elsewhere.
If you use a template, the file that uses the template is automatically an exception.

i.e. what I need is
I want to be able to set which files to exclude when running full lint.

**Comment by pjkaufman** [comment]
I am not sure that I fully grasp your use case. But it does sound like you want to ignore a specific folder. If that is the case, you can specify the folder to not lint in that rule. If you want to specify which files not to lint, you can do that by adding the following in their YAML frontamtter: `disabled rules: [ all ]`. Note that this will not affect custom rules as they are not able to be disabled at this time. But all if not most of the other rules should be able to be disabled in this manner.

Does this satisfy your use case?

**Comment by mineclover** [comment]
no.. 
I wrote this in the above article.
Adding a text to the front matter will also move tags that you deactivated when adding templates to other notes.
I don't want the lint to disappear in other notes to apply the template.
And I want these things to go on automatically.

**Comment by pjkaufman** [comment]
I guess I am not sure what you are looking for then. The Linter doesn't let you specify a list of file names to ignore. Trying to do that would be the equivalent of adding the value to the front matter minus the custom rules being run.

So I don't think I understand what you are proposing would be used to identify which files to ignore.

**Comment by pjkaufman** [comment]
I plan to close this at the the end of the week if I do not hear some more specifics on the desired functionality additions. I am not sure I am understanding what has been provided here and cannot do anything with it at this time. If I better understand this, then I may be to implement it.

**Comment by mineclover** [comment]
sorry for coming late

The template function is used a lot in Obsidian.
Due to the nature of Obsidian, templates are also included in one vault.

If I use lint all files in the vault lint goes in the templates folder as well

I am using auto-modify, create-date create lint
![image](https://user-images.githubusercontent.com/61359316/214773598-2f2f22c8-c380-42cb-9055-4fdff217684c.png)

**Comment by pjkaufman** [comment]
There is a folder level exclusion option called Folders to Ignore.

**Comment by mineclover** [comment]
yes... However, I was lost because there was no example of how to write it.

**Comment by pjkaufman** [comment]
Here is how I ignore them in the setting:
![Screenshot_20230126-134532_Obsidian](https://user-images.githubusercontent.com/29898683/214922984-7688a656-dccb-4d82-9abb-069686b5ef66.jpg)

It is the folder name on its own line in the text area

**Comment by pjkaufman** [comment]
Does that screenshot provide you what you were looking for @mineclover ? If not, what else are you looking for?

**Comment by mineclover** [comment]
![image](https://user-images.githubusercontent.com/61359316/215259141-8db356f6-0da3-453d-bb18-ef5ca72e8ff8.png)
I was ignored when I did it this way
![image](https://user-images.githubusercontent.com/61359316/215259156-34de210c-2622-48d3-aa57-5741ca4af7a5.png)
However, this method failed to ignore the Template folder.

**Comment by pjkaufman** [comment]
What is your folder structure like? The name of the folders is relative to the base of your Obsidin directory.

This could be a bug around the use of Chinese characters in the file path. Can you provide a small example of this for me to test out what is wrong (for example a folder within another folder like what you are doing with those Chinese characters so I can test this myself and see if it is a bug)?

I am not sure what a guide would look like for this as most of the time the template folder or other folders ro ignore are at the root of the service. I would need to check the exact logic, but a relative path should still work.

**Comment by mineclover** [comment]
00-지식관리/001-옵시디언/Template 
This path is the path from vault to Template

**Comment by pjkaufman** [comment]
It looks like I need to test https://github.com/platers/obsidian-linter/blob/master/src/main.ts#L294 with nesting and just make sure it works.

**Comment by mineclover** [comment]
And the code you sent is a code that returns true when file starts with the input path.

It seems that the logic cannot use relative paths.

What I was wondering about was solved
I also found out by testing to separate arrays with enter.

**Comment by mineclover** [comment]
what i was wondering
If you want to use folder ignore
1. Create a path to ignore as an absolute path based on the vault
2. If you want to ignore additionally, separate with enter

this
yes good

**Comment by pjkaufman** [comment]
Ok. I will address the 2 points where possible and see if you have any suggestions.

> 1. Create a path to ignore as an absolute path based on the vault

This is relative to the base of the Obsidian data directory so no absolute paths. This is the general way this is done in Obsidian plugins, but if we need to we can add some kind of clarification in the UI. I would be open to having that info added, but I would need to know what that would look like in terms of additional wording.

> 2. If you want to ignore additionally, separate with enter

This is kind of mentioned in the rule description which says put newlines between folders. I am open to suggestions on how to make this clearer.

Any ideas around these 2 things?
Here is the current info mentioned for Folder Ignore:
``` plaintext
Folders to ignore when linting all files or linting on save. Enter folder paths separated by newlines
```

**Comment by mineclover** [comment]
Yes.. that's a great summary
It would be nice if you could show me a simple example like any other code.
I solved it... but got lost before I could even solve it.

**Comment by yimingli** [comment]
+1 that an example of setting up folders to ignore would be nice. My vault has the following folder structure: `journals/daily`, `journals/monthly`, etc. My goal is to ignore all folders under `journals/`, but I couldn't seem to make it work. I tried the following in the ignore field, but linter still applies to the files under the daily folder:

```
daily
journals/daily
journals
```

**Comment by pjkaufman** [comment]
I am not sure that any of those are invalid folder paths. It may be that trying to escape nested folders is not working, but ignoring a folder should escape all nested folders.

**Comment by pjkaufman** [comment]
I will see about adding a folder to ignore example. I am not sure it has been tested really well, but I can see about adding an example or 2.

**Comment by pjkaufman** [comment]
I will also need to see about converting the ignore option to a folder path packer. Hopefully that will happen this coming release, but we will see how things pan out.

**Comment by pjkaufman** [comment]
It has been a little while, but I am finally revisiting this. I did not like the approach I was taking before so I scrapped part of it and swapped to something more akin to the custom commands suggester. It however just applies to existing folders. I believe it is good to go. However I do need to update the documentation for this for when the next release happens and finish up the word choice. Once I add those 2 things. I will push up the settings changes and close this PR with it getting merged.

Sorry about the delays along the way.
